addons/mrp_subcontracting/models/mrp_production.py

```diff
@@ -128,5 +128,13 @@ def _update_finished_move(self):
 
     def _subcontracting_filter_to_done(self):
         """ Filter subcontracting production where composant is already recorded and should be consider to be validate """
-        mos = self.filtered(lambda mo: mo.state not in ('done', 'cancel'))
-        return mos.filtered(lambda pro: all(line.lot_id for line in pro.move_raw_ids.filtered(lambda sm: sm.has_tracking != 'none').move_line_ids))
+        def filter_in(mo):
+            if mo.state in ('done', 'cancel'):
+                return False
+            if float_is_zero(mo.qty_producing, precision_rounding=mo.product_uom_id.rounding):
+                return False
+            if not all(line.lot_id for line in mo.move_raw_ids.filtered(lambda sm: sm.has_tracking != 'none').move_line_ids):
```

Comment:
Maybe this ? To make shorter and avoid the `filtered`:
```suggestion
            if not all(line.tracking == 'none' or line.lot_id for line in mo.move_line_raw_ids):
```